conans/util/windows.py

```diff
@@ -74,7 +75,14 @@ def path_shortener(path, short_paths):
         # cmd can fail if trying to set ACL in non NTFS drives, ignoring it.
         pass
 
-    redirect = hashed_redirect(short_home, path)
+    if not short_home_env:
+        effective_path = path
+    else:
+        # Strip the home prefix from the path as we don't want it to contribute to the hash
+        from conans.paths import get_conan_user_home
+        home = get_conan_user_home()
+        effective_path = os.path.relpath(path, home)
```

Comment:
The tests fail in this line if the path being shortened is not in (the same drive as) home. Reading a bit more the code I see that the path passed in should be in `storage/path` as defined in `conan.conf` (https://docs.conan.io/en/latest/reference/config_files/conan.conf.html) and the code should read:
```effective_path = os.path.relpath(path, conan_storage_home)```

So far I could not find an easy way to find `conan_storage_home` to compute the proper relative path.

Reply:
I changed the code to construct a ClientCache and got the storage path from there although I am not sure that this is the best way to go.